types/index.d.ts

```diff
@@ -467,7 +467,7 @@ export declare namespace Knex {
   interface OnConflictQueryBuilder<TRecord, TResult> {
     ignore(): QueryBuilder<TRecord, TResult>;
     merge(mergeColumns?: (keyof TRecord)[]): QueryBuilder<TRecord, TResult>;
-    merge(data?: Extract<DbRecord<TRecord>, object>): QueryBuilder<TRecord, TResult>;
+    merge(data?: Extract<DbRecord<ResolveTableType<TRecord, 'update'>>, object>): QueryBuilder<TRecord, TResult>;
```

Comment:
@hermansje @lorefnon Original #4256 didn't have `Extract<>` here, which was added later in #4252.
Is it still needed here or it's safe to drop it?